### PR TITLE
DS-370: Number Input

### PR DESCRIPTION
**Pull request by pauzim**

### CODE REVIEW

File: packages/design-system/src/components/InputNumber/InputNumber.interface.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    | "type"
+    | "onChange"
+    | "suffixIcon"
+    | "onIconClick"
+    | "isError"
+  > {
+  minValue?: number;
+  maxValue?: number;
+  step?: number;
+  enableNegatives?: boolean;
```
Comment (piotrekzawada) [inline packages/design-system/src/components/InputNumber/InputNumber.interface.ts:19]: i really don't like the idea of passing boolean as prop - it indicates that there is an if somewhere down the road. I think we can relay on minValue especially it's optional. So let's assume that all input without minValue can be negative

File: packages/design-system/src/components/InputNumber/index.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          actionType === EActionType.INCREMENT ? count + step : count - step;
+        const newValue = defineNewValue(calculatedValue);
+        defaultRef.current.value = newValue;
+        onChange?.(newValue);
+      }
+    };
+
+    return (
+      <>
+        {label && <Input.Text text={label} type="label" />}
```
Comment (piotrekzawada) [inline packages/design-system/src/components/InputNumber/index.tsx:97]: does the `Input.Text` supports `empty:hidden`? If so we don't need this condition

File: packages/design-system/src/components/InputNumber/index.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+              onClick={() => handleClick(EActionType.DECREMENT)}
+              disabled={disabled}
+              className="rounded-b-[4px]"
+            />
+          </div>
+        </Input.BaseInput>
+
+        {supportingText && !isError && (
+          <Input.Text text={supportingText} type="supportingText" />
+        )}
```
Comment (piotrekzawada) [inline packages/design-system/src/components/InputNumber/index.tsx:129]: same as above

File: packages/design-system/src/components/InputNumber/index.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        <div className="min-h-6">
+          {isError && <Input.Text text={errorMessage} type="errorMessage" />}
```
Comment (piotrekzawada) [inline packages/design-system/src/components/InputNumber/index.tsx:131]: same as above

File: packages/design-system/src/components/InputNumber/index.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    <button
+      onClick={onClick}
+      className={classNames(
+        "flex aspect-square w-3 items-center justify-center bg-neutral-100 hover:bg-neutral-300 hover:text-white",
+        "disabled:cursor-not-allowed disabled:hover:bg-neutral-100 disabled:hover:text-neutral-700",
+        className
+      )}
+      disabled={disabled}
+    >
+      <ChevronDownSvg
```
Comment (zuzannanap) [inline packages/design-system/src/components/InputNumber/index.tsx:28]: i think chevron icons are a bit to small (smaller than on figma)

File: packages/design-system/src/components/InputNumber/index.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+      supportingText,
+      label,
+      disabled,
+      ...props
+    },
+    ref
+  ) => {
+    const defaultRef = useRef<HTMLInputElement>(null);
+    const combinedRef = useCombinedRefs(ref, defaultRef);
+    const isError = !!errorMessage;
```
Comment (zuzannanap) [inline packages/design-system/src/components/InputNumber/index.tsx:56]: box-shadow doesn't change for red when there is an error now

File: packages/design-system/src/stories/atoms/InputNumber.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    errorMessage: errorMessageControl,
+  },
+  args: {
+    label: "",
+    supportingText: "",
+    defaultValue: 0,
+    disabled: false,
+    errorMessage: "",
+    step: 1,
+    minValue: 0,
```
Comment (zuzannanap) [inline packages/design-system/src/stories/atoms/InputNumber.stories.tsx:37]: you can't change input value on storybook when you change min or max value (but i guess it's just o storybook problem)

**Review by zuzannanap — COMMENTED** [review]
i

File: packages/design-system/src/components/InputNumber/index.tsx (hunk 6 above)
Comment (pauzim) [reply packages/design-system/src/components/InputNumber/index.tsx:56]: Ah good one! it's the bug in main input component, I'll try to fix it 👀

File: packages/design-system/src/stories/atoms/InputNumber.stories.tsx (hunk 7 above)
Comment (pauzim) [reply packages/design-system/src/stories/atoms/InputNumber.stories.tsx:38]: I've removed `value` control since it's a controlled component 👌

**Review by piotrekzawada — APPROVED** [review]

File: packages/design-system/src/components/InputNumber/index.tsx (hunk 6 above)
Comment (pauzim) [reply packages/design-system/src/components/InputNumber/index.tsx:56]: fixed in https://github.com/elpassion/flounder-2/pull/145

**Pull request merged**